core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java

```diff
@@ -201,11 +324,17 @@ public synchronized void flush() throws IOException {
     DataOutputStream outputStream = mJournalOutputStream.mOutputStream;
     try {
       outputStream.flush();
+      // Since flush has succeeded, it's safe to clear the mEntriesToFlush queue
```

Comment:
Do we need to recover from journal write failures before calling flush? For example, what will happen if there are 5 writes, the last one fails, and then we call flush?

Reply:
This case is handled by the current implementation. Will add a unit test for this.

Reply:
Done.